File: Ejercicios/Ejercicio3/3.java
class Nave 
{
    private String modelo;
    private int capacidadCombustible;
    private int consumoCombustiblePorHora;

    public Nave(String modelo, int capacidadCombustible, int consumoCombustiblePorHora) 
    {
        this.modelo = modelo;
        this.capacidadCombustible = capacidadCombustible;
        this.consumoCombustiblePorHora = consumoCombustiblePorHora;
    }

    public int calcularCombustibleRestante(int duracionVueloHoras) 
    {
        int combustibleNecesario = duracionVueloHoras * consumoCombustiblePorHora;
        int combustibleRestante = capacidadCombustible - combustibleNecesario;
        return combustibleRestante;
    }
}

class Mision 
{
    private String destino;
    private int duracionHoras;
    private int tripulacionNecesaria;

    public Mision(String destino, int duracionHoras, int tripulacionNecesaria) 
    {
        this.destino = destino;
        this.duracionHoras = duracionHoras;
        this.tripulacionNecesaria = tripulacionNecesaria;
    }

    public void simulacion(Nave nave) 
    {
        System.out.println("Simulando mision hacia " + destino + "...");
        if (nave.calcularCombustibleRestante(duracionHoras) >= 0) 
        {
            System.out.println("La nave tiene suficiente combustible para llegar a " + destino);
            System.out.println("Se necesitan " + tripulacionNecesaria + " tripulantes como minimo");
        } 
        else 
        {
            System.out.println("La nave no tiene suficiente combustible para llegar a " + destino);
            System.out.println("!La misión ha fracasado!");
        }
    }
}

public class Main 
{
    public static void main(String[] args) 
    {
        Nave nave1 = new Nave("Aurora", 10000, 1000);
        Mision mision1 = new Mision("Titan", 10, 5);
        Nave nave2 = new Nave("Sunbeam", 5000, 500);
        Mision mision2 = new Mision("Niese 581D", 10000, 200);

        mision2.simulacion(nave2);
        mision1.simulacion(nave1);
    }
}

File: Ejercicios/Ejercicio3/3.cpp
#include <iostream>
#include <string>

using namespace std;

class Nave 
{
private:
    string modelo;
    int capacidadCombustible;
    int consumoCombustiblePorHora;

public:
    Nave(string modelo, int capacidadCombustible, int consumoCombustiblePorHora) 
    {
        this->modelo = modelo;
        this->capacidadCombustible = capacidadCombustible;
        this->consumoCombustiblePorHora = consumoCombustiblePorHora;
    }

    int CombustibleRestante(int duracionVueloHoras) 
    {
        int combustibleNecesario = duracionVueloHoras * this->consumoCombustiblePorHora;
        int combustibleRestante = this->capacidadCombustible - combustibleNecesario;
        return combustibleRestante;
    }
};

class Mision 
{
private:
    string destino;
    int duracionHoras;
    int tripulacionNecesaria;

public:
    Mision(string destino, int duracionHoras, int tripulacionNecesaria) 
    {
        this->destino = destino;
        this->duracionHoras = duracionHoras;
        this->tripulacionNecesaria = tripulacionNecesaria;
    }

    void Simulacion(Nave nave) 
    {
        cout << "Simulando mision hacia " << this->destino << "..." << endl;
        if (nave.CombustibleRestante(this->duracionHoras) >= 0) 
        {
            cout << "La nave tiene suficiente combustible para llegar a " << this->destino << endl;
            cout << "Se necesitan " << this->tripulacionNecesaria << " tripulantes como minimo" << endl;
        } 
        else 
        {
            cout << "La nave no tiene suficiente combustible para llegar a " << this->destino << endl;
            cout << "!La misión ha fracasado!" << endl;
        }
    }
};

int main() 
{
    Nave nave1("Aurora", 10000, 1000);
    Mision mision1("Titan", 10, 5);
    Nave nave2("Sunbeam", 5000, 500);
    Mision mision2("Niese 581D", 10000, 200);

    mision2.Simulacion(nave2);
    mision1.Simulacion(nave1);

    return 0;
}
